### PR TITLE
Make Zipper opaque. Add fromCons and from constructors

### PR DESCRIPTION
**Pull request by AdrianoFerrari**
Basic attempt to make Zipper private (#10). Added two new constructors as well, since it would probably be useful to have these.

### CODE REVIEW

**Comment by AdrianoFerrari** [comment]
Looking forward to discussion, @wernerdegroot . I think making `Zipper` private is an important API change. What would be the next steps to get this (or something similar) merged? Thoughts?

**Comment by wernerdegroot** [comment]
Hi @AdrianoFerrari , thanks for contributing. Although I haven't responded really quickly, I really appreciate you taking the time 👍 

I would like to propose to add a few unit tests for the new functions.

Also, the fact that the `Zipper` constructors are now private is breaking the currently existing tests. Could you take a look at that too? Maybe it's a good idea to add a function to construct it from a `beforeReversed`, `current` and `after` so people can migrate easily? That could be useful for fixing the tests too!

**Comment by AdrianoFerrari** [comment]
Thanks for the feedback!
I'll try to bring the tests in line, and see about that "`beforeReversed`etc." constructor.

**Comment by AdrianoFerrari** [comment]
Fixed the existing tests, still need to add unit tests for *new* functions.

I'm worried about introducing the `fromBeforeReversed` constructor (or whatever we'd call it), because it's bringing back the confusion that opaque zippers were meant to fix.

Anyone that was using the Zipper constructor directly would have been adding a `List.reverse` to the before term. Migration will be as easy as removing that reversal. I'd say that clarifying the docs for the `from` constructor would be enough to ensure smooth migration.

Thoughts?

**Comment by AdrianoFerrari** [comment]
Added new unit tests, and doc clarification for `from` constructor.

File: src/List/Zipper.elm [hunk 1]
```diff
@@ -69,6 +69,20 @@ fromList xs =
...
 
 
+{-| Construct a `Zipper` from the first element and a list. The `Zipper` will focus on the first element.
+-}
+fromCons : a -> List a -> Zipper a
+fromCons x xs =
+    Zipper [] x xs
+
+
+{-| Construct a `Zipper` from before, current, after. The order is preserved, so `(from [1,2,3] 4 [5,6] |> toList) == [1,2,3,4,5,6]`.
```
Comment (wernerdegroot) [inline src/List/Zipper.elm:79]: Nice! The comment should help a lot!

**Review by wernerdegroot — APPROVED** [review]

**Comment by wernerdegroot** [comment]
Thanks for your contribution. I'll release this directly!

**Comment by wernerdegroot** [comment]
Published as 4.0.0! Well done!

**Comment by AdrianoFerrari** [comment]
Yay :tada:!
Thanks for the guidance and feedback.

**Pull request merged**